src/v2i-hub/MapPlugin/src/MapPlugin.cpp

```diff
@@ -304,6 +207,39 @@ int MapPlugin::Main() {
 	return (EXIT_SUCCESS);
 }
 
+string MapPlugin::enum_to_hex_string()
+{
+	sprintf(mapID_buffer, "%04X", tmx::messages::api::mapData);
+	string map_messageID = mapID_buffer;
+	return map_messageID;
+}
+
+string MapPlugin::removeMessageFrame(string &fileContent)
+{
+    string map_messageID = enum_to_hex_string();
+
+    // Check for and remove MessageFrame
+    if (fileContent.size() >= 4 && fileContent.substr(0, 4) == map_messageID)
```

Comment:
We could do that, but we'd still have to have logic to figure out whether the header has an additional 2 bytes or not. The message frame will always contain the message ID, at the beginning so this is a good way to quickly check for it instead of trying and failing to encode the message. I think both ways would work, but I'd rather run this check instead of attempting a blind encoding.